### PR TITLE
update package versions

### PR DESCRIPTION
**Pull request by JeffFessler**
The main reason for this PR is to verify that Wavelets will work with the latest version of SpecialFunctions.
Currently the Project.toml is pinned to old versions that causes issues for me in other packages.
I don't really expect this PR to be accepted "as is" because you might want to continue to support old versions too, but it would be very helpful to at least get SpecialFunctions version updated.
(CompatHelper is helpful for this purpose BTW.  I could make a PR for that github action if you want.)

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "Wavelets"
 uuid = "29a6e085-ba6d-5f35-a997-948ac2efa89a"
 author = ["Gudmundur Adalsteinsson"]
-version = "0.9.3"
+version = "0.9.4"
```
Comment (johnnychen94) [inline Project.toml:4]:
```suggestion
version = "0.9.3"
```

0.9.3 is not released yet: https://github.com/JuliaRegistries/General/blob/3626f1994b11a7872499e6f696f0ca0ed3c96db3/W/Wavelets/Versions.toml

File: Project.toml [hunk 2]
```diff
@@ -12,11 +12,11 @@ SpecialFunctions = "276daf66-3868-5448-9aa4-cd146d93841b"
...
-DSP = "0.5.1, 0.6"
-FFTW = "0.3, 1"
-Reexport = "0.2, 1.0"
-SpecialFunctions = "0.7.1, 0.8, 0.9, 0.10"
-julia = "1"
+DSP = "0.6"
+FFTW = "1"
+Reexport = "0.2, 1"
+SpecialFunctions = "1"
+julia = "1.5"
```
Comment (johnnychen94) [inline Project.toml:19]: All other compat changes are unnecessary to me; dropping old versions can easily shake the entire downstream ecosystem.

```suggestion
DSP = "0.5.1, 0.6"
FFTW = "0.3, 1"
Reexport = "0.2, 1.0"
SpecialFunctions = "0.7.1, 0.8, 0.9, 0.10, 1"
julia = "1"
```

File: .travis.yml [hunk 3]
```diff
@@ -3,10 +3,7 @@ os:
   - osx
   - linux
 julia:
-  - 1.0
-  - 1.1
-  - 1.2
-  - 1.4
+  - 1.5
```
Comment (johnnychen94) [inline .travis.yml:6]: It still worth checking if this package works nicely with the latest LTS version (1.0.5), while `1` means the latest stable version (1.5.3)

```suggestion
  - 1.0
  - 1
```

**Comment by JeffFessler** [comment]
Hi @johnnychen94, thanks for helping out with this.  I am fine with all your suggestions if they can accepted by the owners of Wavelets.jl.  I have been having annoying package versioning issues and Julia keeps complaining specifically about Wavelets.jl and its restriction to old versions of SpecialFunctions.  The only quick solution I could devise was to fork it and then point MIRT to this forked version.
I left a comment on one of the commits about this:
https://github.com/JuliaDSP/Wavelets.jl/commit/c6a2b42e630d7d777fc0b02a8c34e5a220b87704
but maybe I should have submitted an Issue instead?
What do you think is the best way forward?

A general question about the process here.  You are recommending that the Project.toml file list many old versions of packages, but as far as I can tell the CI only tests the latest version.  So how we know it is valid to list all those old versions as being compatible?

File: Project.toml (hunk 1 above)
Comment (JeffFessler) [reply Project.toml:4]: oh, i saw 0.9.3 in the master branch file so i figured i needed to go one beyond that.  i'll undo.

File: .travis.yml (hunk 3 above)
Comment (JeffFessler) [reply .travis.yml:6]: ok, but 1.5 is soooo much better i can't imagine why anyone would use 1.0.x !

**Comment by johnnychen94** [comment]
> but as far as I can tell the CI only tests the latest version. So how we know it is valid to list all those old versions as being compatible?

Generally, we should keep backward-compatibility as much as possible, and we avoid introducing breaking changes to downstream packages. We should do this because downstream packages/users don't need to change their codes; the less human efforts involved, the better. We don't test all possible versions; we don't have enough CI resources for this and we don't yet have the utils to cover it. [Semantic Versioning(SemVer)](https://semver.org/) is a rule to help us reach the "same" status while requiring less CI coverage.

Let's first discuss what should we do when there're new upstream versions.

As a package maintainer, say, of `Wavelets.jl`, he/she should be aware of at least the release notes of upstream packages(e.g., `SpecialFunctions` and judge/evaluate if the upstream changes will affect `Wavelets.jl`.

It can be irrelevant upstream changes. Non-breaking changes are definitely irrelevant, breaking changes may or may not be related. If `Wavelets.jl` is unrelated to upstream changes, then we could safely tag the new version as compatible. Note that old versions are still working here, so there's no need to drop the compatibility here.

It can also be relevant upstream changes. Relevant changes can be upstream API rename, return value changes, or other more complex cases. As a maintainer, he/she should decide

- if it's okay to absorb the changes as compatibility patches inside `Wavelets.jl`, or
- propagate the breaking changes to downstream packages or end-users.

The ideal choice is to absorb all changes inside `Wavelets.jl`. However, from cases to cases, it is not always practical to do this. It's the maintainers' responsibility to make tradeoffs here.

Now, let's take `SpecialFunctions` as an example. Here's what we already know:

- (1) The current version of `Wavelets` is compatible with `SpecialFunctions = 0.7.1, 0.8, 0.9, 0.10`.
- (2) The current version of `Wavelets` is compatible with `SpecialFunctions@1.2.1`

SemVer is useful because we can approximately infer compatibility from it. According to SemVer, patch versions are backward-compatible and have no new functionalities. Hence

- (3) The current version of `Wavelets` is compatible with `SpecialFunctions@1.2.z`

The next question is: do we need new functionalities introduced in `SpecialFunctions@1`? By revisiting the commit history of `Wavelets.jl` and the release notes of `SpecialFunctions` since `SpecialFunctions@0.10`(since `Wavelets` works nicely with `SpecialFunctions@0.10`), we know that we don't. From this, we know

- (4) The current version of `Wavelets` is compatible with `SpecialFunctions@1.y.z`

Note that SemVer is not the philosophers‘ stone, it requires everyone's awareness and efforts in the ecosystem to make the conclusion (3)/(4) convincing. When compatibility bugs occur, the maintainer of `Wavelets.jl` should check which part of the entire dependency chain doesn't follow the SemVer, and submit a patch or file an issue to appropriate places.

File: .travis.yml (hunk 3 above)
Comment (johnnychen94) [reply .travis.yml]: It is, but as a package maintainer, we should respect LTS and backward-compatibility if we can.

**Comment by JeffFessler** [comment]
@johnnychen94 thanks for the detailed explanation.  I have accepted all your suggestions.  Now I hope that the Wavelets package managers will accept this PR so that compatibility with recent SpecialFunctions is supported!

**Comment by JeffFessler** [comment]
Addresses #64.
Also fixes TagBot to be event driven instead of crontab; see https://github.com/JuliaRegistries/TagBot

**Comment by gummif** [comment]
Thanks for that. Looks good to me.

**Pull request merged**